### PR TITLE
Install data and allow environment override to ASPECT_SOURCE_DIR

### PR DESCRIPTION
**Pull request by gassmoeller**
This is a suggested solution to #1824 that installs the data directory and allows the user to override the path of ASPECT_SOURCE_DIR by setting an environment variable. I need this for an installation where the build and source paths are no longer available after installation.

I implemented a first version using a run-time configuration file instead (as suggested in #1824, see first commit in this PR), but that has the disadvantage that it is not always easy to find the configuration file, even if it is installed in the same folder as the executable (e.g. if ASPECT is called from a different directory and the aspect binary is a weak link). Setting an environment variable seems safer.

However, I am not 100% sure about the name the environment variable should have:
- ASPECT_SOURCE_DIR: This is most similar to the define we use inside ASPECT and the default value for our parameters, so I went with this one, even though it is now technically incorrect because it could point either to the install location or the source location.
- reusing ASPECT_DIR: This would be convenient, we already use this environment variable for plugins to find the aspect installation, however all our default data paths are defined as `$ASPECT_SOURCE_DIR/data`, and while that works for the install directory, it does not work if someone sets ASPECT_DIR to the build directory, because the data is never moved to the build directory
- introducing a new variable ASPECT_ROOT, ASPECT_DATA_DIR or similar: Seemed confusing with the overlap with ASPECT_DIR, also ASPECT_DATA_DIR sounds as if it should point to ASPECT_DIR/data, which it should not.

In summary, I went with ASPECT_SOURCE_DIR, because it is the same as the variable we use in the parameter files, but it seems the best of not perfect options.
For now, I expect this to be only relevant for very few use cases.

### CODE REVIEW

**Comment by gassmoeller** [comment]
After discussion with @jdannberg and @bangerth we have decided we should use a new environment variable $ASPECT_DATA_DIR or similar. I will rework the PR and let you know when it is ready for another review.

File: source/utilities.cc [hunk 1]
```diff
@@ -2148,14 +2148,38 @@ namespace aspect
...
+        }
+      else
+      {
+        // Otherwise, use the default define from config.h
+        return Utilities::replace_in_string(location,
+                                            "$ASPECT_DATA_DIR",
+                                            ASPECT_DATA_DIR);
+      }
+
+      // Otherwise, use the default define from config.h
```
Comment (tjhei) [inline source/utilities.cc:2169]: this comment is out of date

**Review by tjhei — APPROVED** [review]

File: source/utilities.cc (hunk 1 above)
Comment (gassmoeller) [reply source/utilities.cc:2162]: In which way? ASPECT_SOURCE_DIR is defined in config.h.

**Comment by gassmoeller** [comment]
I have addressed @tjhei's comment. Like I mentioned above we had discussed to solve this in a better way, but I think the PR in its current form already has advantages (e.g. if the source directory has to be renamed or moved after an installation happened). So maybe we can merge it anyway and I start the bigger rework at the hackathon? This is slightly time sensitive, because @naliboff wants to use this functionality for a tutorial that starts soon.

**Review by tjhei — APPROVED** [review]
